runtime-testsuite/test/org/antlr/v4/test/runtime/typescript/TscRunner.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2012-2017 The ANTLR Project. All rights reserved.
+ * Use of this file is governed by the BSD 3-clause license that
+ * can be found in the LICENSE.txt file in the project root.
+ */
+package org.antlr.v4.test.runtime.typescript;
+
+import org.antlr.v4.test.runtime.*;
+import org.antlr.v4.test.runtime.states.CompiledState;
+import org.antlr.v4.test.runtime.states.GeneratedState;
+
+import java.io.File;
+import java.nio.file.Files;
+
+import static org.antlr.v4.test.runtime.FileUtils.writeFile;
+import static org.antlr.v4.test.runtime.RuntimeTestUtils.isWindows;
+
+public class TscRunner extends RuntimeRunner {
+
+	/* TypeScript runtime is the same as JavaScript runtime */
+	private final static String NORMALIZED_JAVASCRIPT_RUNTIME_PATH = getRuntimePath("JavaScript").replace('\\', '/');
+	private final static String NPM_EXEC = "npm" + (isWindows() ? ".cmd" : "");
+	private final static String TSC_EXEC = "tsc" + (isWindows() ? ".cmd" : "");
+
+	@Override
+	public String getLanguage() {
+		return "TypeScript";
+	}
+
+
+	@Override
+	protected void initRuntime() throws Exception {
+		installTsc();
+		npmLinkRuntime();
+	}
+
+	private void installTsc() throws Exception {
+		Processor.run(new String[] {NPM_EXEC, "--silent", "install", "-g", "typescript"}, null);
```

Comment:
Please remove `--silent` option here and everywhere. Because of it, runner swallows errors and other useful messages. There is no need to hide compiler messages.

Reply:
There is because messages go to the expected output. Errors are not swallowed

Reply:
No. When I run `npm install --silent -g typescript` on my Ubuntu machine, it prints nothing but surprisingly do nothing. But when I run `npm install -g typescript` it prints the following:

```
npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /usr/local/lib/node_modules/typescript
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/typescript'
npm ERR!  [Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/typescript'] {
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'mkdir',
npm ERR!   path: '/usr/local/lib/node_modules/typescript'
npm ERR! }
npm ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ivankochurkin/.npm/_logs/2022-09-13T20_53_54_181Z-debug-0.log

```

Reply:
yeah it requires sudo, which is implicit in the CI

Reply:
but if you remove --silent and is successful it prints all sorts of things, which interfere with the testing

Reply:
But it should run fine also locally, without CI.